packages/wrangler/src/cfetch/internal.ts

```diff
@@ -32,6 +32,28 @@ export async function performApiFetch(
 	addAuthorizationHeaderIfUnspecified(headers, apiToken);
 	addUserAgent(headers);
 
+	const extraHeaders = getCloudflareExtraHeaders();
+	if (extraHeaders !== "") {
```

Comment:
Without the default value below, this can be `if (extraHeaders) {` covering both `undefined` and `""`.